### PR TITLE
Reorganize early-stage docs sections

### LINKED ISSUE #18127: Reevaluate the organization of "Try out Teleport" and "Deploy a Cluster" guides and include a Linux server "try out" guide
Expected behavior:

In this section:

https://goteleport.com/docs/try-out-teleport/

There is no Linux Server option any more. Given that's one of the most popular options we should bring it back.

Teleport 9.0 had it running:

https://goteleport.com/docs/ver/9.0/getting-started/linux-server/

Also I found that `installation` section https://goteleport.com/docs/installation/ is missing any guides on how to get started. E.g. you have installed helm and there is no continuation to the next steps.

Let's fix this, I think this has big impact on our adoption numbers

### PR DESCRIPTION
**Pull request by ptgott**
Closes #18127

Currently, there are two sections of the docs related to the earliest stages of setting up Teleport, "Try out Teleport" and "Deploy a Cluster". While the organization of our top-level docs sections attempts to correspond to the stages a user goes through when setting up Teleport, these combine some tasks that should probably be separated:

- **Choosing an edition and deploying a production cluster**: While the "Deploy a Cluster" section includes our deployment guides as well as our edition guides, user will likely commit to setting up Teleport Cloud, Open Source, or Enterprise before they plan a full HA production deployment.

- **Deploying a production cluster and deploying a small-scale Teleport cluster:** While the "Deploy a Cluster" section includes the Linux Server and Digital Ocean 1-Click guides, these two guides are more for small-scale deployments than full production clusters.

This change improves the early-stage docs experience for new Teleport users by:

Splitting the "Deploy a Cluster" section
----------------------------------------

The "Deploy a Cluster" section includes subsections related to choosing a Teleport edition as well as subsections related to deploying a production cluster. The problem with this approach is that it leaves no room for adding introductory guides to help users with these two purposes.

By splitting the "Deploy a Cluster" section into two top-level sections, one related to choosing an edition and the other to dpeloying to production, we can add introductory pages that put the subsections of these sections in context.

_This change accompanies the work on #16751 to write a general guide to HA Teleport deployments, since the new guide will have a prominent place in the reorganized "Deploy a Cluster" section._

Adding introductory pages
-------------------------

Sections related to getting started with Teleport, "Try out Teleport" and "Deploy a Cluster", were missing introductory pages. This change adds them, and also adds one to the new "Choose an Edition" section.

Moving the Linux Server and Digital Ocean guides
------------------------------------------------

Move these to "Try out Teleport". The Digital Ocean guide didn't fit with the rest of the deployment guides, since it's not about architecting an HA deployment.

This is also a better home for the "Linux Server" guide, so this change moves that as well.

Removing the "getting-started.mdx" page
---------------------------------------

This page is not reachable via the navigation menu.

### CODE REVIEW

**Comment by ptgott** [comment]
This PR addresses part of #19107, since it makes it clearer that the single-instance deployment in the Linux Server/Digital Ocean guides are intended for small-scale demos.

**Comment by ptgott** [comment]
Left to do:
- [ ] Consider moving the Docker guide (https://goteleport.com/docs/management/guides/docker/) into the "Try out Teleport" section as well, since we have Helm guides for production Kubernetes deployments, and I'm not sure how many users can rely on the Docker guide for other containerized deployments in production (e.g., ECS).

File: docs/pages/choose-an-edition/introduction.mdx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+title: "How to Choose a Teleport Edition"
+description: "Read this guide to determine whether to deploy Teleport Enterprise, Teleport Cloud, or Open Source Teleport"
+---
+
+Teleport supports three editions. In this guide, we will help you determine
+which edition is most appropriate for your use case.
+
+## Editions
+
+### Open Source Teleport
```
Comment (zmb3) [inline docs/pages/choose-an-edition/introduction.mdx:11]: I much prefer `Open Source Teleport`, but I think @xinding33 is starting to say "Teleport Community Edition" more nowadays. Let's confirm with him the naming here.

**Review by zmb3 — APPROVED** [review]

File: docs/pages/choose-an-edition/introduction.mdx (hunk 1 above)
Comment (ptgott) [reply docs/pages/choose-an-edition/introduction.mdx:11]: The Core Concepts page says "Teleport Community Edition", but I must have let my muscle memory get in the way here—I'll change it

**Review by stevenGravy — APPROVED** [review]

File: docs/pages/choose-an-edition/introduction.mdx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+Teleport supports three editions. In this guide, we will help you determine
+which edition is most appropriate for your use case.
+
+## Editions
+
+### Teleport Community Edition
+
+We provide a free, open source distribution of Teleport that enables you to get
+secure access to databases, desktops, Kubernetes clusters, and more.
```
[the remark this answers is not in the document]
Comment (ptgott) [reply docs/pages/choose-an-edition/introduction.mdx:14]: I've suggested using "Windows desktops" to be consistent with the rest of the docs:

```
$ grep -REho "Windows (desktops|systems)" docs/pages | sort | uniq -c
  14 Windows desktops
```

File: docs/pages/choose-an-edition/introduction.mdx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+[Try out Open Source Teleport on a Linux
+server](../try-out-teleport/linux-server.mdx). If you would like to take a look
+at the source, visit the [Teleport GitHub
+repository](https://github.com/gravitational/teleport).
+
+### Teleport Enterprise Cloud
+
+Our team at Teleport manages the Auth Service and Proxy Service, and you add
+resources and configure RBAC.
```
[the remark this answers is not in the document]
Comment (ptgott) [reply docs/pages/choose-an-edition/introduction.mdx:24]: I've added more detail here

**Pull request merged**